### ISSUE TITLE
Regression: false positive IDE0059 in 16.4

### ISSUE THREAD

**Issue by huoyaoyuan** [dotnet/roslyn#39755]
**Version Used**: 
Visual Studio 16.3.8, 16.4 Preview 4.0

**Steps to Reproduce**:
```csharp
        void M()
        {
            IntPtr a = (IntPtr)1;
            try
            {
                var b = a;

                if(Some(a))
                    a = IntPtr.Zero; // here
            }
            finally
            {
                if (a != IntPtr.Zero)
                {

                }
            }
        }

        bool Some(IntPtr a) => true;
```

**Expected Behavior**:
No IDE0059 reported, in 16.3.

**Actual Behavior**:
False positive IDE0059 redundant assignment reported, in 16.4.

Only regresses when the consumptions of `b` and `if(Some(a))` are all present.

**Comment by gafter** [comment]
If you discover this is an issue with the compiler's APIs (for example, the region analysis API reporting that the assignment to `a` doesn't "flow out") then please reassign to the compiler team.

**Comment by mavasani** [comment]
Likely duplicate of https://github.com/dotnet/roslyn/issues/39344 - need to confirm
